### PR TITLE
[Instrumentation.Process] Create src/test project files.

### PR DESCRIPTION
**Pull request by Yun-Ting**
## Changes
Working towards: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/issues/447
Creating project skeleton for process instrumentation.

### CODE REVIEW

File: test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp3.1;net6.0</TargetFrameworks>
+    <TargetFrameworks Condition="$(OS) == 'Windows_NT'">$(TargetFrameworks);net462</TargetFrameworks>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry.Exporter.InMemory" Version="1.3.0" />
```
Comment (Kielek) [inline test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj:9]: Please use other variables from: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/build/Common.nonprod.props (InMemory, vs runner, collector).

**Review by Kielek — COMMENTED** [review]
I think that you are missing workflow, like: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/package-Instrumentation.Runtime.yml

File: test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="xunit" Version="$(XUnitPkgVer)" />
+    <PackageReference Include="coverlet.collector" Version="3.1.2">
```
Comment (reyang) [inline test/OpenTelemetry.Instrumentation.Process.Tests/OpenTelemetry.Instrumentation.Process.Tests.csproj:15]: Consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/7386a192f994fc8d2366c18442d665f1248e98f6/test/OpenTelemetry.Instrumentation.Runtime.Tests/OpenTelemetry.Instrumentation.Runtime.Tests.csproj#L9.

**Review by reyang — APPROVED** [review]
LGTM, consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#contributing-a-new-project (either in this PR or a follow up PR).

**Comment by Yun-Ting** [comment]
> LGTM, consider https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md#contributing-a-new-project (either in this PR or a follow up PR).

Thanks for the link! I've added a project readme and updated the component owners doc.

File: src/OpenTelemetry.Instrumentation.Process/OpenTelemetry.Instrumentation.Process.csproj [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFrameworks>netstandard2.0;net462;netcoreapp3.1;net6.0</TargetFrameworks>
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Process/OpenTelemetry.Instrumentation.Process.csproj:3]: nit - unless we know for sure we need all these targets, we can start with netstandard2.0.

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+### Step 2: Enable Process instrumentation
+
+Process instrumentation should be enabled at application startup using the
+`AddProcessInstrumentation` extension on `MeterProviderBuilder`:
+
+```csharp
+using var meterProvider = Sdk.CreateMeterProviderBuilder()
+    .AddProcessInstrumentation()
+    .AddPrometheusExporter()
```
Comment (cijothomas) [inline src/OpenTelemetry.Instrumentation.Process/README.md:31]: nit: need to mention additional nuget installation for prometheus

**Review by cijothomas — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Process Instrumentation for OpenTelemetry .NET
+
+[![NuGet](https://img.shields.io/nuget/v/OpenTelemetry.Instrumentation.Process.svg)](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Process)
+[![NuGet](https://img.shields.io/nuget/dt/OpenTelemetry.Instrumentation.Process.svg)](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Process)
+
+This is an [Instrumentation
+Library](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#instrumentation-library),
+which instruments [.NET Process](https://docs.microsoft.com/dotnet) and
+collect telemetry about Process behavior.
```
Comment (xiang17) [inline src/OpenTelemetry.Instrumentation.Process/README.md:9]:
```suggestion
which instruments [.NET](https://docs.microsoft.com/dotnet) process and
collect telemetry about process behavior.
```

Nit: No need to use upper case for process.

**Review by xiang17 — APPROVED** [review]

File: src/OpenTelemetry.Instrumentation.Process/OpenTelemetry.Instrumentation.Process.csproj (hunk 3 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/OpenTelemetry.Instrumentation.Process.csproj]: removed others

File: src/OpenTelemetry.Instrumentation.Process/README.md (hunk 4 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/README.md]: added installation info

File: src/OpenTelemetry.Instrumentation.Process/README.md (hunk 5 above)
Comment (Yun-Ting) [reply src/OpenTelemetry.Instrumentation.Process/README.md]: updated; thanks!

**Comment by Yun-Ting** [comment]
> I think that you are missing workflow, like: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/.github/workflows/package-Instrumentation.Runtime.yml

Thanks for spotting it; I've added the workflow.

**Review by utpilla — APPROVED** [review]

File: opentelemetry-dotnet-contrib.sln [hunk 6]
```diff
@@ -233,6 +233,10 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentati
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentation.EventCounters.Tests", "test\OpenTelemetry.Instrumentation.EventCounters.Tests\OpenTelemetry.Instrumentation.EventCounters.Tests.csproj", "{DDA355A3-4D75-4F45-9A5E-E93C3EFB9896}"
 EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentation.Process", "src\OpenTelemetry.Instrumentation.Process\OpenTelemetry.Instrumentation.Process.csproj", "{F811262D-D78A-4C4A-8A31-FFC458164BF2}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "OpenTelemetry.Instrumentation.Process.Tests", "test\OpenTelemetry.Instrumentation.Process.Tests\OpenTelemetry.Instrumentation.Process.Tests.csproj", "{61421ACF-5F90-491B-AFB3-14EF12CCA255}"
+EndProject
```
Comment (utpilla) [inline opentelemetry-dotnet-contrib.sln:239]: nit: Update .sln file to add the package workflow.

**Comment by reyang** [comment]
image

**Comment by utpilla** [comment]
> image

This test has been failing intermittently in the past few days.

**Pull request merged**